ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java

```diff
@@ -103,14 +106,18 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
 
     Object reqId = requestContext.getRequest().getId();
 
+    final Optional<BlockHeader> maybeParentHeader =
+        protocolContext.getBlockchain().getBlockHeader(blockParam.getParentHash());
+
     LOG.atTrace()
         .setMessage("blockparam: {}")
         .addArgument(() -> Json.encodePrettily(blockParam))
         .log();
 
-    var forkValidationResult = validateForkSupported(reqId, blockParam);
-    if (forkValidationResult.isPresent()) {
-      return forkValidationResult.get();
+    ValidationResult<NewPayloadValidationResult> forkValidationResult =
+        validateForkSupported(reqId, blockParam);
+    if (!forkValidationResult.isValid()) {
+      return new JsonRpcErrorResponse(reqId, UNSUPPORTED_FORK);
```

Comment:
This is an improvement in readability IMO and good reuse of the ValidationResult. Not really for this PR, but wonder if we should move ValidationResult to a more generic package since it's already being used outside of the ethereum.mainnet